src/RDom.cpp

```diff
@@ -205,6 +205,16 @@ RDom::operator RVar() const {
     return x;
 }
 
+void RDom::where(Expr predicate) {
+    user_assert(!dom.frozen())
+        << (*this) << " cannot be given a new predicate, because it has already"
+        << " been used in the update definition of some function.\n";
+    if (!dom.defined()) {
```

Comment:
For consistency, how about:

    user_assert(dom.defined()) << "Error: Can't add predicate to undefined RDom.\n"

instead of if(!dom.defined()) { ... }